eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -326,7 +325,6 @@ public synchronized BackupRegistry get() {
 
         this.backupRegistryProvider = backupRegistryProvider;
 
-        this.urlRandomizer = new EndpointUtils.InstanceInfoBasedUrlRandomizer(instanceInfo);
```

Comment:
Let's keep the urlRandomizer as a configurable aspect of the DiscoveryClient and not static state of ResolverUtils. 
 This can be made customizable either via Guice bindings or configuration.  I'm also thinking this should be done as a ServiceUrlRandomizerFactory that takes an InstanceInfo argument.